core/src/main/java/org/stellar/anchor/event/models/TransactionEvent.java

```diff
@@ -0,0 +1,141 @@
+package org.stellar.anchor.event.models;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.gson.annotations.SerializedName;
+import java.time.Instant;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+@AllArgsConstructor
+public class TransactionEvent implements AnchorEvent {
+  @SerializedName("event_id")
+  String eventId;
+
+  Type type;
+
+  public String getType() {
+    return this.type.type;
+  }
+
+  String id;
+
+  Sep sep;
+
+  Kind kind;
+
+  Status status;
+
+  @SerializedName("amount_expected")
+  Amount amountExpected;
+
+  @SerializedName("amount_in")
+  Amount amountIn;
+
+  @SerializedName("amount_out")
+  Amount amountOut;
+
+  @SerializedName("amount_fee")
+  Amount amountFee;
+
+  @SerializedName("quote_id")
+  String quoteId;
+
+  @SerializedName("started_at")
+  Instant startedAt;
+
+  @SerializedName("updated_at")
+  Instant updatedAt;
+
+  @SerializedName("completed_at")
+  Instant completedAt;
+
+  @SerializedName("transfer_received_at")
+  Instant transferReceivedAt;
+
+  String message;
+
+  Refund refund;
+
+  @SerializedName("stellar_transactions")
+  StellarTransaction[] stellarTransactions;
+
+  @SerializedName("external_transaction_id")
+  String externalTransactionId;
+
+  @SerializedName("custodial_transaction_id")
+  String custodialTransactionId;
+
+  @SerializedName("source_account")
+  String sourceAccount;
+
+  @SerializedName("destination_account")
+  String destinationAccount;
+
+  Customers customers;
+
+  StellarId creator;
+
+  public enum Status {
+    PENDING_SENDER("pending_sender"),
+    PENDING_STELLAR("pending_stellar"),
+    PENDING_CUSTOMER_INFO_UPDATE("pending_customer_info_update"),
+    PENDING_TRANSACTION_INFO_UPDATE("pending_transaction_info_update"),
+    PENDING_RECEIVER("pending_receiver"),
+    PENDING_EXTERNAL("pending_external"),
+    COMPLETED("completed"),
+    ERROR("error");
+
+    @JsonValue public final String status;
+
+    Status(String status) {
+      this.status = status;
+    }
+  }
+
+  public enum Sep {
+    SEP_31(31);
+
+    private final Integer sep;
+
+    Sep(Integer sep) {
+      this.sep = sep;
+    }
+
+    @JsonValue
+    public Integer getSep() {
+      return sep;
+    }
+  }
+
+  public enum Type {
+    TRANSACTION_CREATED("transaction_created"),
+    TRANSACTION_PAYMENT_RECEIVED("transaction_payment_received"),
```

Comment:
@lijamie98 has just updated this event name to `transaction_status_changed` in https://github.com/stellar/stellar-anchor-platform/pull/31:
```suggestion
    TRANSACTION_STATUS_CHANGED("transaction_status_changed"),
```

Reply:
lets address that in a separate PR since it requires a schema change